[vrisdng] iP

vrisdng opened the pull request:
### AmiyaBot

> The cute, gentle assistant that every Dokutah loves, who will remind you to grind your daily missions.

- text-based
- easy to learn
- fast to use
- and can speak Japanese!

Download instruction:
1. Clone this [repo](https://github.com/vrisdng/ip) to your local
2. Run Amiya
3. Add your tasks
4. Let Amiya manage your tasks for you!

**Features:**

- [x] Managing tasks
- [x] Managing deadlines and events
- [x] Speak in Japanese with you 😉

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

vrisdng committed:
Level-0: Rename, Greet, Exit

vrisdng committed:
Level-1: Echo

vrisdng committed:
Level-2: Add,List

vrisdng committed:
Level-3: Mark as Done

vrisdng committed:
Level-4: ToDos, Events, Deadlines

vrisdng committed:
Level-5: A-TextUiTesting

vrisdng committed:
Level-5: Handle Errors

vrisdng committed:
Level-6: Delete

vrisdng committed:
Add enums for task types

vrisdng committed:
Save

vrisdng committed:
Add Amiya.txt file

vrisdng committed:
Level-8: Dates and Time

vrisdng committed:
A-MoreOOP

vrisdng committed:
A-Packages

vrisdng committed:
Add an Amiya image

vrisdng committed:
Update README.md

vrisdng committed:
Merge branch 'add-gradle-support'

vrisdng committed:
Add Graddle

vrisdng committed:
Add JUnit Tests

vrisdng committed:
Add JavaDoc comments

vrisdng committed:
Follow the Google Coding Standard

vrisdng committed:
Level-9: Find

vrisdng committed:
Place better line break

vrisdng committed:
Merge branch 'branch-A-CodingStandard'

vrisdng committed:
Update Amiya.java

vrisdng committed:
Merge branch 'branch-Level-9'

juneha1120 commented on src/main/java/amiya/ui/UI.java:
```diff
@@ -105,6 +105,27 @@ public void showTasks(List<Task> tasks) {
+     * @param tasks The list of tasks that match the keyword.
+     */
+
+    public void showFoundTasks(List<Task> tasks) {
```
I like this clear and self-explanatory verb naming for this method!

juneha1120 commented on src/main/java/amiya/task/Task.java:
```diff
@@ -101,9 +102,9 @@ public void unmark() {
     public static Task fromFileFormat(String line) {
         String[] parts = line.split("\\|");
-        if (parts.length < 3) return null;
+        if (parts.length < 3)
```
Shouldn't the curly brackets be put here?

juneha1120 commented on src/main/java/Amiya.java:
```diff
@@ -1,24 +1,35 @@
+    private static Storage storage;
+    private static TaskList taskList;
+    private static UI ui;
+    private static Parser parser;
```
I like these clear naming of variables!

juneha1120 commented on src/main/java/amiya/task/TaskList.java:
I like how all methods are named in standard language with appropriate lengths

KarKuen commented on src/main/java/amiya/ui/UI.java:
```diff
@@ -0,0 +1,197 @@
+     *
+     * @param name The name to be included in the greeting.
+     */
+    public void greeting(String name) {
```
```suggestion
    public void giveGreetings(String name) {
```

KarKuen commented on src/main/java/amiya/ui/UI.java:
```diff
@@ -0,0 +1,197 @@
+     * @param command The command to translate.
+     * @return The translated string or the original command if no translation is found.
+     */
+    public String translateToJapanese(String command) {
```
cute 👍

KarKuen commented on src/main/java/amiya/task/TaskList.java:
```diff
@@ -0,0 +1,154 @@
+
+    public Task removeTask(int taskId) throws AmiyaException {
+        if (tasks.isEmpty()) {
+            throw new AmiyaException("there are no tasks to remove.");
```
I like this Exception that you came up with

KarKuen commented on src/main/java/amiya/Amiya.java:
```diff
@@ -0,0 +1,113 @@
+            command = scanner.nextLine().trim();
+            String[] parts = command.split("\\s+");
+            String commandType = parts[0];
+            try {
+                if (command.equalsIgnoreCase("bye")) {
+                    break;
+                } else if (command.equalsIgnoreCase("list")) {
+                    ui.showTasks(taskList.getTasks());
+                } else if (commandType.equalsIgnoreCase("echo") && parts.length > 1) {
+                    String textToEcho = command.substring(5).trim();
+                    ui.echo(textToEcho);
+                } else if (commandType.equalsIgnoreCase("find") && parts.length > 1) {
+                    String keyword = command.substring(5).trim();
+                    ui.showFoundTasks(taskList.findTasks(keyword));
+                } else if (commandType.equals("mark")) {
+                    taskList.markTask(Integer.parseInt(parts[1]));
+                    ui.showTaskMarked(taskList.getTasks().get(Integer.parseInt(parts[1]) - 1));
+                } else if (commandType.equals("unmark")) {
+                    taskList.unmarkTask(Integer.parseInt(parts[1]));
+                    ui.showTaskUnmarked(taskList.getTasks().get(Integer.parseInt(parts[1]) - 1));
+                } else if (commandType.equals("remove")) {
+                    Task task = taskList.removeTask(Integer.parseInt(parts[1]));
+                    ui.showTaskDeleted(task, taskList.size());
+                } else if (commandType.equals("clear")) {
+                    taskList.clearTasks();
+                    ui.showTasksCleared();
+                } else if (commandType.equals("todo") || commandType.equals("deadline")
+                        || commandType.equals("event")) {
+                    handleTasks(command);
+                } else {
+                    throw new AmiyaException("this is an unknown command.");
+                }
+                storage.save(taskList.getTasks());
+            } catch (AmiyaException | NumberFormatException e) {
+                ui.showInvalidCommand(e.getMessage());
```
Should this be extracted out?

KarKuen reviewed (commented):
Overall, your code is clean and easy understand. Good job :3

vrisdng committed:
Remove bars

vrisdng committed:
Fix coding style of Ui class

vrisdng committed:
Add checkstyle

vrisdng committed:
Set up basic FXML views

juneha1120 reviewed (commented):
Overall I like how your code is neat and clear to understand!

vrisdng committed:
Finish the basic GUI

vrisdng committed:
Extract commands logic

vrisdng committed:
Merge branch 'branch-Level-10'

vrisdng committed:
Update GUI

Refactor Amiya response and exit flow for GUI integration

The system was previously unable to show Amiya's responses for more complex inputs
on the GUI and the exit message was not
displayed before the application closed.

Let's
* change the return type of Ui methods to strings,
so that getResponse can directly use these
* modify the init of Amiya
* use asynchronous programming in Java to
guarantee the message is shown before exiting.

This approach enhances the user experience by ensuring a smooth transition
and proper display of the exit message before application termination.

vrisdng committed:
Add assertions to TaskList and Deadline

The current code does not ensure the integrity of task data and lacks proper validation for critical operations like adding, removing, marking, and unmarking tasks.

This needs to change to avoid potential issues such as invalid task indices, null descriptions, and inconsistent dates.

Let's:
* add assertions to TaskList class and Deadline class to:
- Validate non-null and non-empty descriptions when adding a task.
- Ensure valid task indices for all task-related operations.
- Check that the task list is not empty before marking or unmarking tasks.
- Validate that descriptions and times are non-null and non-empty.

Assertions are used to catch potential bugs early in the development process, ensuring the task data remains in a valid and expected state.

These changes improve error handling and ensure better consistency in task management.

vrisdng committed:
Extract parseTask from load() method in Storage

vrisdng committed:
Merge pull request #1 from vrisdng/branch-A-Assertions

Add assertions to TaskList and Deadline

vrisdng committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

vrisdng committed:
Merge pull request #2 from vrisdng/branch-A-CodeQuality

Extract parseTask from load() method in Storage

vrisdng committed:
Add ScheduleCommand

vrisdng committed:
Merge branch 'branch-update-GUI'

vrisdng committed:
Clean up Amiya.java

vrisdng committed:
Merge branch 'branch-Extension'

vrisdng committed:
Add the new command to the command list

vrisdng committed:
Add guide for users in-app

vrisdng committed:
Add more JUnit tests for Deadline, Event, Parser, Storage, and more commands

vrisdng committed:
Update CommandTest.java

vrisdng committed:
Update README.md

vrisdng committed:
Change CSS

vrisdng committed:
Merge branch 'master' of https://github.com/vrisdng/ip

vrisdng committed:
Update README.md

vrisdng committed:
Update README.md

vrisdng committed:
Merge branch 'branch-A-MoreTesting'

vrisdng committed:
Merge branch 'master' of https://github.com/vrisdng/ip

vrisdng committed:
Update README.md

vrisdng committed:
Update README.md

vrisdng committed:
Throw exceptions in parsing commands that require more descriptions

vrisdng committed:
Merge branch 'master' of https://github.com/vrisdng/ip

vrisdng committed:
Add UserGuide

vrisdng committed:
Add the images in the UG

vrisdng committed:
Add Javadoc comments to more methods and improve README
